### PR TITLE
Add empty host and trailing slash to windows file paths

### PR DESCRIPTION
**Pull request by phil-davis**
This is a rebase of PR #25 with conflicts resolved...

In the 2nd commit, I have added some comments and simplified the windows_path regex in a similar way to what @peterpostmann has suggested in comment https://github.com/sabre-io/uri/pull/25#discussion_r946370109

### CODE REVIEW

**Comment by phil-davis** [comment]
@peterpostmann please have a look and comment if you agree or not with what I have done here.

File: .php-cs-fixer.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...aces":true,"no_blank_lines_after_class_opening":true,"no_leading_import_slash":true,"no_whitespace_in_blank_line":true,"ordered_class_elements":{"order":["use_trait"]},"return_type_declaration":true,"short_scalar_cast":true,"single_blank_line_before_namespace":true,"single_trait_insert_per_statement":true,"ternary_operator_spaces":true,"visibility_required":true,"blank_line_after_namespace":true,"constant_case":true,"elseif":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_space_around_double_colon":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true},"hashes":{"tests\/Uri\/BuildTest.php":1670421601,"tests\/Uri\/SplitTest.php":1630284748,"tests\/Uri\/ParseTest.php":2861228548,"tests\/Uri\/NormalizeTest.php":3826344400,"tests\/Uri\/ResolveTest.php":258490035,"lib\/Version.php":479674545,"lib\/InvalidUriException.php":1408242133,"lib\/functions.php":2265006986}}
```
Comment (staabm) [inline .php-cs-fixer.cache]: this file should be git ignored?

**Comment by peterpostmann** [comment]
I found one more issue: We are not catching the case, where it's just the drive letter, i.e. ```file:///C:```
Do you want to make the whole part after the colon optional? ```/^(?<windows_path> [A-Z]:([\/\\\\].*)?)$/x```

File: .php-cs-fixer.cache (hunk 1 above)
Comment (phil-davis) [reply .php-cs-fixer.cache]: yes, my mistake. That is added to `.gitignore` in PR #74 - and I had one accidentally lying around from working on that PR. I will remove it from here.

**Comment by phil-davis** [comment]
> I found one more issue: We are not catching the case, where it's just the drive letter, i.e. `file:///C:` Do you want to make the whole part after the colon optional? `/^(?<windows_path> [A-Z]:([\/\\\\].*)?)$/x`

done in 3rd commit. The new test case fails before the change and passes with the change.

**Comment by phil-davis** [comment]
IMO we are good-to-go here. @staabm reviewed the original PR previously. There has been commentary from people above that indicates they are OK with the direction this is going.

I will merge and release a patch version. That will let me move forward with PR #74 that will cause another release.

**Comment by phil-davis** [comment]
Also see discussion in issue #81 PR #25 and issue #31

**Pull request merged**